packages/mgt-chat/src/utils/rewriteEmojiContent.ts

```diff
@@ -30,3 +30,68 @@ const processEmojiContent = (messageContent: string): string => {
  */
 export const rewriteEmojiContent = (content: string): string =>
   emojiMatch(content) ? processEmojiContent(content) : content;
+
+/**
+ * The emoji HTML string that an <emoji> tag is translated to.
+ *
+ * @param emoji used as the alt value.
+ * @param title used in titles.
+ * @param id used in URLs and id attributes,
+ * @param hasOtherContent determines which size of the emoji to fetch. default 50px.
+ * @returns HTML string.
+ */
+const emojiHtmlString = (emoji: string, title: string, id: string, hasOtherContent?: boolean): string => {
+  const size = hasOtherContent ? '20' : '50'; // 20px with content, 50px without.
+  return `<span contenteditable="false" title="${title}" type="(${id})" class="animated-emoticon-${size}-${id}"><img itemscope="" itemtype="http://schema.skype.com/Emoji" itemid="${id}" src="https://statics.teams.cdn.office.net/evergreen-assets/personal-expressions/v2/assets/emoticons/${id}/default/${size}_f.png" title="${title}" alt="${emoji}" style="width:${size}px;height:${size}px;"/></span>`;
+};
+
+/**
+ * Matches the emoji tag with id in group 2, alt in group 3 and title in group 4.
+ */
+const emojiRegexForHtml =
+  /(<emoji[^>]+)id=["'](\w*[^"']*)["']\s+alt=["'](\w*[^"']*)["']\s+title=["'](\w*[^"']*)["']><\/emoji>/;
+
+/**
+ * Perform a check on content and determine whether there is other content
+ * apart from <emoji> tags.
+ * @param content the entire message content.
+ * @param hasAttachments killswitch if there are attachments for the content.
+ * @returns true if there is other content other than <emoji> tags in the content string.
+ */
+const checkForOtherContent = (content: string, hasAttachments: boolean): boolean => {
```

Comment:
I don't think that we need the `hasAttachments`  parameter, or if we do decide to pass it in then we should exit early if it's true. I can't see a need to do the regex work if we have attachments